pkg/observability/resource.go

```diff
@@ -72,6 +73,12 @@ func NewStackdriverMonitoredResource(ctx context.Context, c *StackdriverConfig)
 	}
 	labels["location"] = region
 
+	// Metadata often returns the following: "projects/111111111111/regions/us-central1"
+	pieces := strings.Split(region, "/")
```

Comment:
This feels fragile and could have subtle bugs if that format changes

Reply:
Yeah. I'm not sure what to do otherwise. I feel like a regex would be more fragile?

Reply:
SplitN and checking against the other lengths? Or maybe returning/logging an error on mismatch?

Reply:
good idea, added error message.